src/main/java/uk/gov/pay/api/service/CreatePaymentService.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.pay.api.service;
+
+import org.apache.http.HttpStatus;
+import uk.gov.pay.api.app.config.PublicApiConfig;
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.exception.CreateChargeException;
+import uk.gov.pay.api.model.ChargeFromResponse;
+import uk.gov.pay.api.model.CreatePaymentRequest;
+import uk.gov.pay.api.model.links.PaymentWithAllLinks;
+import uk.gov.pay.api.utils.JsonStringBuilder;
+
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static javax.ws.rs.client.Entity.json;
+
+public class CreatePaymentService {
+    private final String baseUrl;
+    private final Client client;
+    private final PublicApiUriGenerator publicApiUriGenerator;
+    private final ConnectorUriGenerator connectorUriGenerator;
+
+    @Inject
+    public CreatePaymentService(Client client, PublicApiConfig configuration, PublicApiUriGenerator publicApiUriGenerator, ConnectorUriGenerator connectorUriGenerator) {
+        this.client = client;
+        this.baseUrl = configuration.getBaseUrl();
+        this.publicApiUriGenerator = publicApiUriGenerator;
+        this.connectorUriGenerator = connectorUriGenerator;
+    }
+
+    public PaymentWithAllLinks invoke(Account account, CreatePaymentRequest requestPayload) {
```

Comment:
i think `create` would be a better method name

Reply:
Renamed from `invoke` to `create`